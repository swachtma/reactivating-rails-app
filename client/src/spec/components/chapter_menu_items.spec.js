/* global expect, jest */
import React from 'react'
import { shallow } from 'enzyme'

import configEnzyme from '../../test_helper'
import ChapterMenuItems from '../../components/chapter_menu_items'

const test_props = {
  chapters: [
    { id: 1, title: 'Test Chapter 1' },
    { id: 2, title: 'Test Chapter 2' }
  ],
  active_chapter: { id: 2, title: 'Test Chapter 1' },
  dispatchRouteChapter: jest.fn(),
  handleSidebarVisibility: jest.fn()
}

let w = shallow(<ChapterMenuItems {...test_props} />)

describe('<ChapterMenuItems />', () => {
  it('renders according to snapshot', () => expect(w).toMatchSnapshot())

  it('calls handleSidebarVisibility on focus', () => {
    w.simulate('focus')
    expect(test_props.handleSidebarVisibility).toBeCalled()
  })

  it('call handleSidebarVisibility on focus', () => {
    w.simulate('change')
    expect(test_props.dispatchRouteChapter).toBeCalled()
  })

  it('disables link to active chapter', () => {
    expect(w.prop('options')[1].active).toBe(true)
  })

  it('prints each chapter as a link', () => {
    expect(w.prop('options').length).toBe(2)
  })
})
