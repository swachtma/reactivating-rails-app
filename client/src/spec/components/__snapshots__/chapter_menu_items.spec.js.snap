// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChapterMenuItems /> renders according to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChapterMenuItems
    active_chapter={
        Object {
            "id": 2,
            "title": "Test Chapter 1",
          }
    }
    chapters={
        Array [
            Object {
              "id": 1,
              "title": "Test Chapter 1",
            },
            Object {
              "id": 2,
              "title": "Test Chapter 2",
            },
          ]
    }
    dispatchRouteChapter={[Function]}
    handleSidebarVisibility={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "additionLabel": "Add ",
      "additionPosition": "top",
      "className": "link item",
      "closeOnBlur": true,
      "deburr": false,
      "icon": "dropdown",
      "minCharacters": 1,
      "noResultsMessage": "No results found.",
      "onChange": [Function],
      "onFocus": [Function],
      "openOnFocus": true,
      "options": Array [
        Object {
          "active": false,
          "id": "chapter_menu_item_1",
          "key": "chapter_1",
          "text": "Test Chapter 1",
          "value": 1,
        },
        Object {
          "active": true,
          "id": "chapter_menu_item_2",
          "key": "chapter_2",
          "text": "Test Chapter 2",
          "value": 2,
        },
      ],
      "placeholder": "Jump to Chapter...",
      "pointing": "top right",
      "renderLabel": [Function],
      "searchInput": "text",
      "selectOnBlur": false,
      "selectOnNavigation": false,
      "style": Object {
        "margin": "0 1em",
      },
      "tabIndex": "1",
      "text": "Test Chapter 1",
      "value": 2,
      "wrapSelection": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "additionLabel": "Add ",
        "additionPosition": "top",
        "className": "link item",
        "closeOnBlur": true,
        "deburr": false,
        "icon": "dropdown",
        "minCharacters": 1,
        "noResultsMessage": "No results found.",
        "onChange": [Function],
        "onFocus": [Function],
        "openOnFocus": true,
        "options": Array [
          Object {
            "active": false,
            "id": "chapter_menu_item_1",
            "key": "chapter_1",
            "text": "Test Chapter 1",
            "value": 1,
          },
          Object {
            "active": true,
            "id": "chapter_menu_item_2",
            "key": "chapter_2",
            "text": "Test Chapter 2",
            "value": 2,
          },
        ],
        "placeholder": "Jump to Chapter...",
        "pointing": "top right",
        "renderLabel": [Function],
        "searchInput": "text",
        "selectOnBlur": false,
        "selectOnNavigation": false,
        "style": Object {
          "margin": "0 1em",
        },
        "tabIndex": "1",
        "text": "Test Chapter 1",
        "value": 2,
        "wrapSelection": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
